src/shared/Core/ApplicationBase.cs

```diff
@@ -75,6 +77,31 @@ public Task<int> RunAsync(string[] args)
                 Context.Trace.WriteLine("Tracing of secrets is enabled. Trace output may contain sensitive information.");
             }
 
+            // Enable TRACE2 tracing
+            if (Context.Settings.GetTrace2TracingEnabled(out string trace2Value))
+            {
+                var connectionPath = PlatformUtils.IsWindows() ? trace2Value.TrimUntilLastIndexOf("/") :
+                    trace2Value.TrimUntilIndexOf(":");
+
+                if (!string.IsNullOrEmpty(connectionPath))
+                {
+                    Context.Trace2.AddListener(new CollectorWriter(
+                            () => new NamedPipeClientStream(".", connectionPath, PipeDirection.Out,
```

Comment:
I'm confused.  Does NamedPipeClientStream() talk to Unix domain sockets on Unix??
I would have expected this section to be guarded with platform selectors.

Reply:
The implementation of the `NamedPipe*Stream` classes on Unix is always UDS, yes. They are also limited to 'byte' mode.

Reply:
ok thanks! i'll keep that in mind. i think we're ok if i always do readline() rather than assuming i always get a whole message.